app/models/user.rb

```diff
@@ -1,7 +1,15 @@
 class User < ActiveRecord::Base
+  include HasRole
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
-  devise :database_authenticatable, 
+  devise :database_authenticatable,
          :recoverable, :rememberable, :trackable, :validatable
-  belongs_to :role, polymorphic: true
+  belongs_to :role, polymorphic: true, dependent: :destroy
```

Comment:
`dependent: :destroy` should be added to a few more associations, like the gift requests under a list, but I feel like this PR isn't the place to do it.